pkg/node/manager/manager.go

```diff
@@ -449,6 +475,29 @@ func (m *Manager) NodeUpdated(n nodeTypes.Node) {
 	}
 }
 
+// deleteIPCache deletes the IP addresses from the IPCache with the 'oldSource'
+// if they are not found in the newIPs slice.
+func (m *Manager) deleteIPCache(oldSource source.Source, oldIPs []net.IP, newIPs []string) {
+	for _, address := range oldIPs {
+		if address == nil {
+			continue
+		}
+		addrStr := address.String()
+		var found bool
+		for _, healthIPAdded := range newIPs {
+			if healthIPAdded == addrStr {
```

Comment:
nit: Seems like a more general name, such as `ipAdded` would be better than `healthIPAdded` here?

Reply:
:+1: